### ISSUE TITLE
500 Server Error

### ISSUE THREAD

**Issue by alessioanghileri** [arc298/instagram-scraper#585]
is:issue is:open It seems something changed in Instagram api today: during a common request with --tag, I receive the following error: "500 Server Error: Internal Server Error for url: https://www.instagram.com/graphql/query/?***".
The same request was fine till yesterday.
Any idea what's going on?

**Comment by eo4929** [comment]
I'm also encountering this issue ..;;;

**Comment by rennanharo** [comment]
Me too. Any ideas? Tried to use a proxy server to make the requests but no success either.

**Comment by spainnoobperez** [comment]
My symptoms resembles to yours, but my error is 403.

**Comment by simone-deola** [comment]
It seems to be a problem caused by the introduction of Reels on the Instagram platforms. The initial calls and response to IG have been changed and it seems that the IG team is still working on this changes.

**Comment by rennanharo** [comment]
> It seems to be a problem caused by the introduction of Reels on the Instagram platforms. The initial calls and response to IG have been changed and it seems that the IG team is still working on this changes.
@simone-deola 

Thanks for the information. Is there any source to that? I want to know first-hand when these changes complete.

**Comment by simone-deola** [comment]
> > It seems to be a problem caused by the introduction of Reels on the Instagram platforms. The initial calls and response to IG have been changed and it seems that the IG team is still working on this changes.
> > @simone-deola
> 
> Thanks for the information. Is there any source to that? I want to know first-hand when these changes complete.

No, my information are just experimental information. I had notice that there are some new Reel variables in the package exchanged with IG. 
Also the yesterday 500 error disappear in the browser communication, but instagram-scraper does not working as before and still reply with a 500 error. 

I've fixed the error in the instagram-scraper by replacing 
QUERY_HASHTAG = BASE_URL + 'graphql/query/?query_hash=ded47faa9a1aaded10161a2ff32abb6b&variables={0}' and
QUERY_HASHTAG_VARS = '{{"tag_name":"{0}","first":50,"after":"{1}"}}' with
QUERY_HASHTAG = BASE_URL + 'graphql/query/?query_hash=9b498c08113f1e09617a1703c22b2f32&variables={0}' and
QUERY_HASHTAG_VARS = '{{"tag_name":"{0}","first":50,"after":"{1}"}}' in the constants.py file.

It's seem that the query_hash in the request have been changed in this two days. I'm not an expert, I've just copied the behavior seen in the browser. I dunno what I've done :)

**Comment by rennanharo** [comment]
> > > It seems to be a problem caused by the introduction of Reels on the Instagram platforms. The initial calls and response to IG have been changed and it seems that the IG team is still working on this changes.
> > > @simone-deola
> > 
> > 
> > Thanks for the information. Is there any source to that? I want to know first-hand when these changes complete.
> 
> No, my information are just experimental information. I had notice that there are some new Reel variables in the package exchanged with IG.
> Also the yesterday 500 errors disappear in the browser communication, but instagram-scraper does not working as before and still replies with a 500 error.
> 
> I've fixed the error in the instagram-scraper by replacing
> QUERY_HASHTAG = BASE_URL + 'graphql/query/?query_hash=ded47faa9a1aaded10161a2ff32abb6b&variables={0}' and
> QUERY_HASHTAG_VARS = '{{"tag_name":"{0}","first":50,"after":"{1}"}}' with
> QUERY_HASHTAG = BASE_URL + 'graphql/query/?query_hash=9b498c08113f1e09617a1703c22b2f32&variables={0}' and
> QUERY_HASHTAG_VARS = '{{"tag_name":"{0}","first":50,"after":"{1}"}}' in the constants.py file.
> 
> It's seem that the query_hash in the request have been changed in this two days. I'm not an expert, I've just copied the behavior seen in the browser. I dunno what I've done :)

Oh, that's great. I'll be testing it out to see if is addresses the issue and come back later with some feedback. Nevertheless, thanks a lot!

**Comment by rennanharo** [comment]
> > > It seems to be a problem caused by the introduction of Reels on the Instagram platforms. The initial calls and response to IG have been changed and it seems that the IG team is still working on this changes.
> > > @simone-deola
> > 
> > 
> > Thanks for the information. Is there any source to that? I want to know first-hand when these changes complete.
> 
> No, my information are just experimental information. I had notice that there are some new Reel variables in the package exchanged with IG.
> Also the yesterday 500 errors disappear in the browser communication, but instagram-scraper does not working as before and still replies with a 500 error.
> 
> I've fixed the error in the instagram-scraper by replacing
> QUERY_HASHTAG = BASE_URL + 'graphql/query/?query_hash=ded47faa9a1aaded10161a2ff32abb6b&variables={0}' and
> QUERY_HASHTAG_VARS = '{{"tag_name":"{0}","first":50,"after":"{1}"}}' with
> QUERY_HASHTAG = BASE_URL + 'graphql/query/?query_hash=9b498c08113f1e09617a1703c22b2f32&variables={0}' and
> QUERY_HASHTAG_VARS = '{{"tag_name":"{0}","first":50,"after":"{1}"}}' in the constants.py file.
> 
> It's seem that the query_hash in the request have been changed in this two days. I'm not an expert, I've just copied the behavior seen in the browser. I dunno what I've done :)

Worked like a charm. I'm no expert either, and have no clue what this change does. All I know is that it worked and that I appreciate it a lot. Thanks!

**Comment by jasebell** [comment]
> Worked like a charm. I'm no expert either, and have no clue what this change does. All I know is that it worked and that I appreciate it a lot. Thanks!

Worked a treat for me, thanks @simone-deola !!
